sql/sqlserver/server_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tobias Schottdorf
+
+package sqlserver
+
+import (
+	"reflect"
+	"strconv"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/sql/sqlwire"
+)
+
+func TestProcessSelect(t *testing.T) {
+	s := (*Server)(nil) // enough for now
+
+	vInt := int64(5)
+	vNum := "3.14159"
+	vStr := "two furs one cub"
+	vBool := true
+
+	dInt := sqlwire.Datum{IntVal: &vInt}
+	dFloat := func() sqlwire.Datum {
```

Comment:
Am I correct in guessing that this is needed to avoid a shadowing warning? We should decide on a go-to pattern when we need to deal with cases like this. I can think of a few options:
- use a closure like you've done here
- do the `if-else` contortion i've used in a handful of places in #1736
- rename the innermost `err`

Of course the final option: fix (or fork) `vet`.